feat(practice): add dashboard components and setup flow for practice …

vincentxuu opened the pull request:
…management

- Introduced new components: CheckInView, DashboardFlow, HistoryView, MainDashboard, and related shared components for managing practice sessions.
- Implemented a setup flow for creating new practices, including steps for inputting goals and resources.
- Enhanced context and service layers to support new features and data handling.
- Updated ESLint configuration to disable specific accessibility rule for labels.

vincentxuu committed:
feat(practice): add dashboard components and setup flow for practice management

- Introduced new components: CheckInView, DashboardFlow, HistoryView, MainDashboard, and related shared components for managing practice sessions.
- Implemented a setup flow for creating new practices, including steps for inputting goals and resources.
- Enhanced context and service layers to support new features and data handling.
- Updated ESLint configuration to disable specific accessibility rule for labels.

vincentxuu committed:
fix(Practice/Setup): correct formatting in StepTwo component

JohnsonMao commented on services/practice/checkIn.ts:
`service` 這邊的寫法，可以請 AI 參考 `service/modules/circle` 的寫法，
另外把 practice 這個模組放到 `service/modules` 下面比較好做管理。

JohnsonMao commented on services/practice/index.ts:
index.ts 只用來做檔案的匯出，可參考 `/service/modules` 資料夾下面串 API 的方式實作

JohnsonMao commented on contexts/PracticeContext.tsx:
請使用 useSWR 的方式拿取資料，並用自訂義 hook 的方式，封裝操作邏輯，不需要再定義 Context。
自定義的 hook 可以放到 `feature/practice` 的資料夾下面

JohnsonMao commented on components/Practice/Shared/ProgressBar.tsx:
```diff
@@ -0,0 +1,82 @@
+      {/* 進度條 */}
+      <div className={`w-full bg-gray-200 rounded-full ${sizeClasses[size]} overflow-hidden`}>
+        <div
+          className={`${sizeClasses[size]} rounded-full transition-all duration-500 ease-out ${
```
請使用 `utils/cn` 的 `cn` 來串接 tailwindCSS 的樣式

JohnsonMao commented on components/Practice/Shared/StatusBadge.tsx:
```diff
@@ -0,0 +1,58 @@
+
+  const getIcon = () => {
+    const iconProps = {
+      className: size === 'sm' ? 'h-3 w-3' : size === 'lg' ? 'h-5 w-5' : 'h-4 w-4'
```
嵌套的三元運算子可讀性與維護性不太好，建議用物件的方式定義

JohnsonMao commented on components/Practice/Setup/StepFour.tsx:
```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+import { IoCheckmarkOutline as Check, IoCalendarOutline as Calendar, IoFlagOutline as Target, IoBookOutline as BookOpen, IoNotificationsOutline as Bell, IoPeopleOutline as Users } from 'react-icons/io5';
+import { colors, contentTypeOptions, getUnitType } from '@/constants/practice';
+import { PathInfo } from '../../../services/practice';
```
可以用 `@/services/practice` 的方式引入，盡量避免多個 ../

JohnsonMao commented on components/Practice/Shared/Confetti.tsx:
```diff
@@ -0,0 +1,87 @@
+  const uniqueId = uuidv4();
+
+  // 生成紙屑碎片
+  const pieces = Array(100).fill(0).map(() => {
```
請把 `Array(100).fill(0)` ，改成 `Array.from({ length: 100 }, (_, i) => i)`，並定義在 Components 外面，避免每次重新渲染都要重新建立長度 100 的 Array，造成無謂的性能消耗

JohnsonMao commented on services/practice/types.ts:
```diff
@@ -0,0 +1,268 @@
+  getProgress: (practiceId: string) => number;
+  canCheckInToday: (practiceId: string) => boolean;
+
+  // 便利方法（向後相容）
```
這不是在開發新功能嗎？為什麼要向後兼容

JohnsonMao commented on services/practice/utils.ts:
這裡的有些工具現有套件都有實作了，例如：`uuid` 可以產 id、`dayjs` 可以處理日期、進度計算這些應該封裝到 hook 或 components 中用來處理顯示邏輯

JohnsonMao reviewed (changes requested):
即個影響可讀性的地方需要修改
1. 不要嵌套三元運算子
2. tailwindCSS 的 className 樣式請用 `cn` 來串接
3. 不需要沒有意義的註釋
4. 請明確定義型別，避免使用 `any`

JohnsonMao commented:
對了～ StepOne, StepTwo,...
要不要改成 Step1, Step2,...
這樣 Components 在資料夾中也比較集中

然後也可以考慮將這個模組的都放到 features 這個資料夾中

vincentxuu committed:
feat(practice): integrate lucide-react icons and refactor components

- Added `lucide-react` as a dependency for icon usage across components.
- Refactored `StatusBadge`, `PracticeCard`, and other components to utilize icons from `lucide-react`.
- Removed unused components and context files to streamline the codebase.
- Introduced new hooks and services for better practice management and data handling.
- Updated the setup flow and dashboard components to enhance user experience.

vincentxuu commented:
/gemini review

vincentxuu committed:
Merge branch 'dev' into feat/practice

vincentxuu committed:
feat(practice): add tags and daily goal settings to practice schema and components

vincentxuu commented:
/gemini review

vincentxuu committed:
chore: update ESLint configuration and refine linting script to include additional directories

vincentxuu committed:
fix: correct parameter naming and formatting in Milestones context and practice components

vincentxuu commented:
/gemini review

vincentxuu committed:
refactor(practice): streamline hooks and components, remove unused types, and enhance practice management functionality

vincentxuu committed:
feat(practice): implement scroll restoration and enhance practice hooks for improved navigation and user experience

vincentxuu commented:
/gemini review

vincentxuu committed:
Merge branch 'dev' into feat/practice

vincentxuu committed:
refactor(practice): restructure service imports and enhance utility functions

vincentxuu committed:
Merge branch 'dev' into feat/practice

vincentxuu committed:
fix(DeleteConfirm): change Badge variant from destructive to default for improved UI clarity

vincentxuu committed:
feat(practice): enhance progress bar and mood options, add learning insights component

vincentxuu committed:
Merge branch 'dev' into feat/practice

vincentxuu committed:
fix(DeleteConfirm): update button variant from destructive to alert for better user experience

vincentxuu committed:
chore(featureFlags): remove unnecessary blank line for cleaner code

vincentxuu commented:
/gemini review
